### PR TITLE
CLOUD-3171 add cookie jar for websocket client to handle liveid cookie

### PR DESCRIPTION
**Pull request by MaximeHeckel**
@tifayuki can you also take a quick look to this?

### CODE REVIEW

File: dockercloud/stream.go [hunk 1]
```diff
@@ -49,8 +63,15 @@ func dial() (*websocket.Conn, *http.Response, error) {
 	header.Add("Authorization", AuthHeader)
 	header.Add("User-Agent", customUserAgent)
 
+	if GetCookieByName(cookies, "liveid") != "" {
+		header.Add("Cookie", "liveid="+GetCookieByName(cookies, "liveid"))
```
Comment (tifayuki) [inline dockercloud/stream.go]: can you run `go fmt` before committing?

**Comment by tifayuki** [comment]
Why didn't we add all the cookie jar like the rest APIs, which also includes `device_id`?

File: dockercloud/stream.go [hunk 2]
```diff
@@ -49,8 +63,15 @@ func dial() (*websocket.Conn, *http.Response, error) {
...
+	}
+
 	var Dialer websocket.Dialer
-	return Dialer.Dial(Url, header)
+
+	ws, res, err := Dialer.Dial(Url, header)
+
+	return ws, res, err
```
Comment (MaximeHeckel) [inline dockercloud/stream.go]: I wanted to add something in between and thought it was clearer

**Comment by MaximeHeckel** [comment]
I looked for a cookie jar implementation for the websocket client on Gorilla websocket but couldn't find any.
Maybe there is, but if there isn't I might submit a PR to the repo of the package

**Comment by MaximeHeckel** [comment]
https://github.com/gorilla/websocket/pull/166 when this PR is merged we can merge this one

**Comment by fermayo** [comment]
@tifayuki review again please

**Pull request merged**